### PR TITLE
fix(build): Align pull invocation based on correct package size

### PR DESCRIPTION
**Pull request by craciunoiuc**
### Prerequisite checklist

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [x] Ran `make fmt` on your commit series before opening this PR;
  - [ ] Updated relevant documentation.

### CODE REVIEW

File: cmd/kraft/build/build.go [hunk 1]
```diff
@@ -310,6 +310,9 @@ func (opts *Build) Run(cmd *cobra.Command, args []string) error {
 	if len(missingPacks) > 0 {
 		for _, p := range missingPacks {
 			p := p // loop closure
+			if newLen := len(unikraft.TypeNameVersion(p)) + len("pulling "); !norender && newLen > nameWidth {
+				nameWidth = newLen
+			}
```
Comment (nderjung) [inline cmd/kraft/build/build.go]: I'd add simply use the integer value to decrease processing time and a comment indicating that it refers to additional length of the string used by the left text name.

Also I believe "`configuring `" is the longest string used.

Comment (nderjung) [reply cmd/kraft/build/build.go]: Having said this, shouldn't this be added to [the pre-emptive calculation](https://github.com/unikraft/kraftkit/blob/staging/cmd/kraft/build/build.go#L134-L138)? The issue is probably here?

Comment (nderjung) [reply cmd/kraft/build/build.go]: The calculation there is wrong because it does not take into account the version.

Comment (craciunoiuc) [reply cmd/kraft/build/build.go]: Not 100% sure why, but in the other cases it works (fetching), I can change there and try out again

It's used only in 2 places anyways

File: cmd/kraft/build/build.go [hunk 2]
```diff
@@ -127,12 +127,17 @@ func (opts *Build) Run(cmd *cobra.Command, args []string) error {
...
+		}
+
 		// The longest word is "configuring" (which is 11 characters long), plus
 		// additional space characters (2 characters), brackets (2 characters) the
 		// name of the project and the target/plat string (which is variable in
 		// length).
-		for _, targ := range project.Targets() {
-			if newLen := len(targ.Name()) + len(target.TargetPlatArchName(targ)) + 15; newLen > nameWidth {
+		for _, component := range comps {
+			if newLen := len(unikraft.TypeNameVersion(component)) + 15; newLen > nameWidth {
```
Comment (antoineco) [inline cmd/kraft/build/build.go]: Can we please add a `const` that describes the value `15`?

```go
const maxLengthOfBlah = 15  // or equivalent
```

Comment (antoineco) [reply cmd/kraft/build/build.go]: (there is a comment about it above so this is more like a nice to have, not a blocking comment)

Comment (craciunoiuc) [reply cmd/kraft/build/build.go]: 🖖

Comment (craciunoiuc) [reply cmd/kraft/build/build.go]: done 💪

**Review by antoineco — APPROVED** [review]

**Pull request merged**